google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerImpl.java

```diff
@@ -1067,18 +1061,21 @@ ResultSet executeQueryInternalWithOptions(
           new ResumableStreamIterator(MAX_BUFFERED_CHUNKS, QUERY) {
             @Override
             CloseableIterator<PartialResultSet> startStream(@Nullable ByteString resumeToken) {
-              return new CloseableServerStreamIterator<PartialResultSet>(
+              GrpcStreamIterator stream = new GrpcStreamIterator(prefetchChunks);
+              SpannerRpc.StreamingCall call =
                   rpc.executeQuery(
                       resumeToken == null
                           ? request
                           : request.toBuilder().setResumeToken(resumeToken).build(),
-                      null,
-                      session.options));
-
-              // TODO(hzyi): make resume work
```

Comment:
Sorry I'm losing track of work items. Does resume work now?

Reply:
This todo item was added because previous PR switched from `SpannerRpc.StreamingCall` to `ServerStream` which was not able to handle resuming as Spanner expected without non-trivial work. This PR switched back to `SpannerRpc.StreamingCall` (the code is quite similar to [master](https://github.com/GoogleCloudPlatform/google-cloud-java/blob/86ea052196d793637480d3ff1fad7a3856425d1f/google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/SpannerImpl.java#L1170)) except that we are now using `StreamController` to control the stream instead of [`ResultStreamConsumer`](https://github.com/GoogleCloudPlatform/google-cloud-java/blob/27c90f1298bb27a618ff59b3cd17321c577f3a6b/google-cloud-clients/google-cloud-spanner/src/main/java/com/google/cloud/spanner/spi/v1/SpannerRpc.java#L127) so resume should automatically work.

Reply:
LGTM